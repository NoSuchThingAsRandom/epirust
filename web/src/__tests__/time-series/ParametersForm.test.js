/*
 * EpiRust
 * Copyright (c) 2020  ThoughtWorks, Inc.
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with this program.  If not, see <http://www.gnu.org/licenses/>.
 *
 */

import React from 'react'
import { render, fireEvent } from '@testing-library/react'
import ParametersForm from '../../time-series/ParamtersForm'
import renderer from 'react-test-renderer'

test('should render ParametersForm with defaults', () => {
    const component = renderer.create(<ParametersForm onDataSubmit={jest.fn()} />)
    let tree = component.toJSON()

    expect(tree).toMatchSnapshot()
})

test('invoke onsubmit handler passed on form submit', () => {
    const handleSubmitData = jest.fn()
    const { getByTestId } = render(<ParametersForm onDataSubmit={handleSubmitData} />)

    fireEvent.submit(getByTestId('simulationForm'))

    expect(handleSubmitData).toHaveBeenCalledTimes(1)
})

test('invoke handler on form submit', () => {
    const handleSubmitData = jest.fn()
    const { getByTestId } = render(<ParametersForm onDataSubmit={handleSubmitData} />)
    const expectedData = {
        "death_rate": 0.2,
        "disease_name": "small_pox",
        "emergency_workers_population": 0.1,
        "grid_size": 250,
        "high_transmission_rate": 0.5,
        "high_transmission_start_day": 16,
        "hospital_spread_rate_threshold": 100,
        "last_day": 22,
        "lockdown_at_number_of_infections": 100,
        "number_of_agents": 10000,
        "public_transport_percentage": 0.2,
        "regular_transmission_rate": 0.05,
        "regular_transmission_start_day": 10,
        "simulation_hrs": 10000,
        "vaccinate_at": 5000,
        "vaccinate_percentage": 0.2,
        "working_percentage": 0.7,
    }

    fireEvent.submit(getByTestId('simulationForm'))
    expect(handleSubmitData).toHaveBeenCalledWith(expectedData)
})

