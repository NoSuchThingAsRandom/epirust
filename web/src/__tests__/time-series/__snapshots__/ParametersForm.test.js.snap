// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ParametersForm with defaults 1`] = `
<form
  className="user-inputs"
  data-testid="simulationForm"
  onSubmit={[Function]}
>
  <div
    className="form-row"
  >
    <div
      className="col inputs"
    >
      <div>
        <fieldset>
          <legend>
            Population
          </legend>
          <div
            className="input-control"
          >
            <label
              className="col-form-label-sm"
              htmlFor="number_of_agents"
            >
              Number of Agents
            </label>
            <input
              aria-describedby="number_of_agents"
              className="form-control form-control-sm"
              defaultValue="10000"
              id="number_of_agents"
              name="number_of_agents"
              placeholder="Number of Agents"
              type="number"
            />
          </div>
          <div
            className="input-control"
          >
            <label
              className="col-form-label-sm"
              htmlFor="public_transport_percentage"
            >
              Public Transport Percentage
            </label>
            <input
              aria-describedby="public_transport_percentage"
              className="form-control form-control-sm"
              defaultValue="0.2"
              id="public_transport_percentage"
              name="public_transport_percentage"
              placeholder="Public Transport Percentage"
              step="any"
              type="number"
            />
          </div>
          <div
            className="input-control"
          >
            <label
              className="col-form-label-sm"
              htmlFor="working_percentage"
            >
              Working Percentage
            </label>
            <input
              aria-describedby="working_percentage"
              className="form-control form-control-sm"
              defaultValue="0.7"
              id="working_percentage"
              max="1"
              min="0"
              name="working_percentage"
              placeholder="Working Percentage"
              step="any"
              type="number"
            />
          </div>
        </fieldset>
        <fieldset>
          <div
            className="input-control"
          >
            <label
              className="col-form-label-sm"
              htmlFor="grid_size"
            >
              Grid Size
            </label>
            <input
              aria-describedby="grid_size"
              className="form-control form-control-sm"
              defaultValue="250"
              id="grid_size"
              name="grid_size"
              placeholder="Grid Size"
              type="number"
            />
          </div>
          <div
            className="input-control"
          >
            <label
              className="col-form-label-sm"
              htmlFor="simulation_hrs"
            >
              Simulation Hours
            </label>
            <input
              aria-describedby="simulation_hrs"
              className="form-control form-control-sm"
              defaultValue="10000"
              id="simulation_hrs"
              name="simulation_hrs"
              placeholder="Simulation Hours"
              type="number"
            />
          </div>
        </fieldset>
      </div>
      <fieldset>
        <legend>
          Interventions
        </legend>
        <div
          className="input-control"
        >
          <label
            className="col-form-label-sm"
            htmlFor="vaccinate_at"
          >
            Vaccinate At
          </label>
          <input
            aria-describedby="vaccinate_at"
            className="form-control form-control-sm"
            defaultValue="5000"
            id="vaccinate_at"
            name="vaccinate_at"
            placeholder="Vaccinate At"
            type="number"
          />
        </div>
        <div
          className="input-control"
        >
          <label
            className="col-form-label-sm"
            htmlFor="vaccinate_percentage"
          >
            Vaccinate Percentage
          </label>
          <input
            aria-describedby="vaccinate_percentage"
            className="form-control form-control-sm"
            defaultValue="0.2"
            id="vaccinate_percentage"
            name="vaccinate_percentage"
            placeholder="Vaccinate Percentage"
            step="any"
            type="number"
          />
        </div>
        <div
          className="input-control"
        >
          <label
            className="col-form-label-sm"
            htmlFor="lockdown_at_number_of_infections"
          >
            Vaccinate Percentage
          </label>
          <input
            aria-describedby="lockdown_at_number_of_infections"
            className="form-control form-control-sm"
            defaultValue="100"
            id="lockdown_at_number_of_infections"
            name="lockdown_at_number_of_infections"
            placeholder="Lockdown At(number of infections)"
            step="any"
            type="number"
          />
        </div>
        <div
          className="input-control"
        >
          <label
            className="col-form-label-sm"
            htmlFor="emergency_workers_population"
          >
            Emergency Workers Population
          </label>
          <input
            aria-describedby="emergency_workers_population"
            className="form-control form-control-sm"
            defaultValue="0.1"
            id="emergency_workers_population"
            name="emergency_workers_population"
            placeholder="Emergency Workers Population"
            step="any"
            type="number"
          />
        </div>
        <div
          className="input-control"
        >
          <label
            className="col-form-label-sm"
            htmlFor="hospital_spread_rate_threshold"
          >
            Hospital Spread Rate Threshold
          </label>
          <input
            aria-describedby="hospital_spread_rate_threshold"
            className="form-control form-control-sm"
            defaultValue="100"
            id="hospital_spread_rate_threshold"
            name="hospital_spread_rate_threshold"
            placeholder="Hospital Spread Rate Threshold"
            step="any"
            type="number"
          />
        </div>
      </fieldset>
      <fieldset>
        <legend>
          Disease Dynamics
        </legend>
        <div
          className="input-control"
        >
          <label
            className="col-form-label-sm"
            htmlFor="disease_name"
          >
            Disease Name
          </label>
          <select
            aria-describedby="disease_name"
            className="form-control form-control-sm"
            data-testid="select-disease"
            defaultValue="small_pox"
            id="disease_name"
            name="disease_name"
            onChange={[Function]}
            placeholder="Disease Name"
          >
            <option
              value="small_pox"
            >
              Small Pox
            </option>
            <option
              value="sars"
            >
              SARS
            </option>
            <option
              value="covid_19"
            >
              COVID-19
            </option>
          </select>
        </div>
        <div
          className="input-control"
        >
          <label
            className="col-form-label-sm"
            htmlFor="regular_transmission_start_day"
          >
            Regular Transmission Start Day
          </label>
          <input
            aria-describedby="regular_transmission_start_day"
            className="form-control form-control-sm"
            defaultValue={10}
            id="regular_transmission_start_day"
            name="regular_transmission_start_day"
            placeholder="Regular Transmission Start Day"
            type="text"
          />
        </div>
        <div
          className="input-control"
        >
          <label
            className="col-form-label-sm"
            htmlFor="high_transmission_start_day"
          >
            High Transmission Start Day
          </label>
          <input
            aria-describedby="high_transmission_start_day"
            className="form-control form-control-sm"
            defaultValue={16}
            id="high_transmission_start_day"
            name="high_transmission_start_day"
            placeholder="High Transmission Start Day"
            type="text"
          />
        </div>
        <div
          className="input-control"
        >
          <label
            className="col-form-label-sm"
            htmlFor="last_day"
          >
            Last Day
          </label>
          <input
            aria-describedby="last_day"
            className="form-control form-control-sm"
            defaultValue={22}
            id="last_day"
            name="last_day"
            placeholder="Last Day"
            type="text"
          />
        </div>
        <div
          className="input-control"
        >
          <label
            className="col-form-label-sm"
            htmlFor="regular_transmission_rate"
          >
            Regular Transmission Rate
          </label>
          <input
            aria-describedby="regular_transmission_rate"
            className="form-control form-control-sm"
            defaultValue={0.05}
            id="regular_transmission_rate"
            name="regular_transmission_rate"
            placeholder="Regular Transmission Rate"
            type="text"
          />
        </div>
        <div
          className="input-control"
        >
          <label
            className="col-form-label-sm"
            htmlFor="high_transmission_rate"
          >
            High Transmission Rate
          </label>
          <input
            aria-describedby="high_transmission_rate"
            className="form-control form-control-sm"
            defaultValue={0.5}
            id="high_transmission_rate"
            name="high_transmission_rate"
            placeholder="High Transmission Rate"
            type="text"
          />
        </div>
        <div
          className="input-control"
        >
          <label
            className="col-form-label-sm"
            htmlFor="death_rate"
          >
            Death Rate
          </label>
          <input
            aria-describedby="death_rate"
            className="form-control form-control-sm"
            defaultValue={0.2}
            id="death_rate"
            name="death_rate"
            placeholder="Death Rate"
            type="text"
          />
        </div>
      </fieldset>
    </div>
    <div
      className="col actions"
    >
      <button
        className="btn btn-primary"
        id="submitBtn"
        type="submit"
      >
        Start
      </button>
    </div>
  </div>
</form>
`;
